torch/_inductor/kernel/mm.py

```diff
@@ -107,7 +107,14 @@
             b = tl.load(B, mask=rk[:, None] < k, other=0.)
         if B_PROLOGUE_CAST_TYPE is not None:
             b = b.to(B_PROLOGUE_CAST_TYPE)
-        acc += tl.dot(a, b, allow_tf32=ALLOW_TF32)
+        input_precision = "tf32" if ALLOW_TF32 else "ieee"
```

Comment:
I would just augment ALLOW_TF32 to be named "matmul strategy" and do the processing logic outside of the template.